src/routing.js

```diff
@@ -0,0 +1,76 @@
+import {chain, merge, toPairs, trim} from 'lodash';
+
+const isParam = segment => segment.substr(0, 1) === ':';
+const toSegments = pathname => trim(pathname, '/').split('/');
+
+const buildRoute = (segments, methods) => {
+  const segment = segments.shift();
+  const edge = segments.length === 0;
+
+  return {
+    [segment]: merge(
+      { $param: isParam(segment) },
+      edge
+        ? { $edge: edge, $methods: methods }
+        : buildRoute(segments, methods)
+    )
+  };
+};
+
+const buildParams = (param, value) =>
+  param ? { params: { [param.substr(1)]: value } } : {};
+
+const lookup = (segments, routes) => {
```

Comment:
Please add some documentation for this method. What are `segments` and `routes` types and what are the expected return types.